runtime/src/main/java/org/corfudb/runtime/BootstrapUtil.java

```diff
@@ -66,10 +67,26 @@ public static void bootstrap(@Nonnull Layout layout,
                             .addClient(new ManagementHandler())
                             .addClient(new BaseHandler());
 
-                    new LayoutClient(router, layout.getEpoch())
-                            .bootstrapLayout(layout).get();
-                    new ManagementClient(router, layout.getEpoch())
-                            .bootstrapManagement(layout).get();
+                    LayoutClient layoutClient = new LayoutClient(router, layout.getEpoch());
+                    ManagementClient managementClient
+                            = new ManagementClient(router, layout.getEpoch());
+
+                    try {
+                        CFUtils.getUninterruptibly(layoutClient.bootstrapLayout(layout),
+                                AlreadyBootstrappedException.class);
+                    } catch (AlreadyBootstrappedException abe) {
+                        if (!layoutClient.getLayout().get().equals(layout)) {
+                            log.error("Layout Server already bootstrapped with different layout.");
+                            throw abe;
+                        }
+                    }
+
+                    try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)